File: 05_Recursion/How_Recursion_Uses_Stack/main.cpp
#include <stdio.h>
using namespace std;

void fun1(int n){
 if(n>0) {
      printf("%d ",n);
      fun1(n-1);
 }
}


int main(){
 int x=3;
 fun1(x);
}

// int main(){
//  int x=3;
//  fun1(x);
// }

// #include <stdio.h>
// using namespace std;

// void fun2(int n){
//  if(n>0) {
//       fun2(n-1);
//       printf("%d ",n);
    
//  }
// }

File: 05_Recursion/How_Recursion_Uses_Stack/main.go
package main

import "fmt"

func fun1(n int) { // 3,2,1
	if n > 0 {
		fmt.Printf("%d ", n)
		fun1(n - 1)
	}
}

func main() {
	x := 3
	fun1(x)
}

// package main // 1,2,3

// import "fmt"

// func fun2(n int) { // 3,2,1
// 	if n > 0 {
// 		fun2(n - 1)
// 		fmt.Printf("%d ", n)
// 	}
// }

// func main() {
// 	x := 3
// 	fun2(x)
// }
